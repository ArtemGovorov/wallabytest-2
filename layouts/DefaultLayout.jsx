import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import LanguageSelector from '../components/LanguageSelector/LanguageSelector';

const Main = styled.main`
  position: relative;
  max-width: 320px;
  margin: 0 auto;
  overflow: hidden;
`;

class DefaultLayout extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
  };

  state = {
    isLoading: false,
  };

  handleLangChange = isLoading => {
    // To force chatbot a rerender
    this.setState({ isLoading });
  };

  render() {
    const { children } = this.props;
    const { isLoading } = this.state;

    return (
      !isLoading && (
        <div>
          <LanguageSelector onChange={this.handleLangChange} />

          <Main>{children}</Main>
        </div>
      )
    );
  }
}

export default DefaultLayout;
