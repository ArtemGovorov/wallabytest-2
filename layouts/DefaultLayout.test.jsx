import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import DefaultLayout from './DefaultLayout';

const mockChildren = <div>hello world</div>;

describe('<DefaultLayout />', () => {
  const component = shallow(<DefaultLayout>{mockChildren}</DefaultLayout>);
  test('renders as expected', () => {
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
