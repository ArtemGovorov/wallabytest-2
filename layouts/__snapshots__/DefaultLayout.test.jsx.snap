// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DefaultLayout /> renders as expected 1`] = `
<div>
  <LanguageSelector
    onChange={[Function]}
  />
  <styled.main>
    <div>
      hello world
    </div>
  </styled.main>
</div>
`;
